### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by bborgesr**
The goal is that this will be a commit-sane version of https://github.com/rstudio/shinydashboard/pull/187

### CODE REVIEW

**Comment by bborgesr** [comment]
Closes #179 .
Closes #71.
Closes #87.

**Comment by bborgesr** [comment]
Need to look better at #114 and #128, to see if these can now be closed as well..

**Comment by bborgesr** [comment]
Also need to look better at #38 before closing.

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
+
+* Address [#179](https://github.com/rstudio/shinydashboard/issues/179) support for bookmarking the expanded/collapsed state of the whole sidebar. (commit [#e71c93f](https://github.com/rstudio/shinydashboard/pull/199/commits/e71c93fa7a71f229e725efd4a7867e431cd57679))
+
+* Added Shiny input to keep track of which sidebar `menuItem` is expanded (if any), which makes bookmarking the exact state of the sidebar trivial. (commit [#6901b90](https://github.com/rstudio/shinydashboard/pull/199/commits/6901b90b8c866b89d02514cfc01fdfab88175602))
+
+### Minor new features and improvements
+
+* Split JS files. (commit [#ea91503](https://github.com/rstudio/shinydashboard/pull/199/commits/ea915038ae2126f48c15e3aac41782a22b16c506)). More updates to Gruntfile and structure. (commit [#4e80616](https://github.com/rstudio/shinydashboard/pull/199/commits/4e80616c5b3aa0dc73022dc815288b5ba7c35be0))
+
+* Added `findAttribute()` function. (commit [#97b952f](https://github.com/rstudio/shinydashboard/pull/199/commits/97b952f13fc5e6781895d49c40e865d4719cb86a))
```
Comment (wch) [inline NEWS.md]: I don't think you need to mention this, since it's not exported.

File: NEWS.md [hunk 2]
```diff
@@ -1,6 +1,32 @@
 shinydashboard 0.5.3.9000
 =========================
 
+This is a maintenance release of Shinydashboard, mostly aimed at fixing bugs and introducing minor features. 
+
+## Full changelog
+
+### New features
+
+* Address [#179](https://github.com/rstudio/shinydashboard/issues/179) support for bookmarking the expanded/collapsed state of the whole sidebar. (commit [#e71c93f](https://github.com/rstudio/shinydashboard/pull/199/commits/e71c93fa7a71f229e725efd4a7867e431cd57679))
```
Comment (wch) [inline NEWS.md]: For commits, you don't need the `#`, so instead of `#e71c93f`, it would be `e71c93f`.

File: srcjs/AdminLTE/README-shiny-mods.md [hunk 3]
```diff
@@ -3,3 +3,14 @@ This branch of AdminLTE contains the following changes from the stock version, t
 * The box collapse function triggers 'shown' and 'hidden' events, so that Shiny knows when outputs are visible or not. See https://github.com/rstudio/shinydashboard/issues/42 for a test app (must re-apply commit 73f6027 when updating to newer version of Admin LTE).
 
 * In AdminLTE.css, the fonts are fetched from the local host, instead of from Google fonts (must re-apply commits e9e63d1 and 9ccb12d when updating to newer version of Admin LTE).
+
+* Add the following code chunk to app.js (see commit #???):
```
Comment (wch) [inline srcjs/AdminLTE/README-shiny-mods.md]: Does this need to be updated with a ref to the correct commit?

**Comment by wch** [comment]
Other than the small things I commented on, this looks good.

File: R/dashboardSidebar.R [hunk 4]
```diff
@@ -117,12 +117,20 @@ dashboardSidebar <- function(..., disable = FALSE, width = NULL, collapsed = FAL
     '))))
   }
 
+  # If we're restoring a bookmarked app, this holds the value of whether or not the
+  # sidebar was collapsed. If this is not the case, the default is whatever the user
+  # specified in the `collapsed` argument.
+  dataValue <- shiny::restoreInput(id = "sidebarCollapsed", default = collapsed)
+  if (disable) dataValue <- TRUE
```
Comment (wch) [inline R/dashboardSidebar.R]: Maybe a comment here to about the transition issue that this is a workaround for?

**Pull request merged**